lib/ai/model/generationtool.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+)
+
+type commandGenerationTool struct{}
+
+type commandGenerationToolInput struct {
+	// Command is a unix command to execute.
+	Command string `json:"command"`
+}
+
+func (c *commandGenerationTool) Name() string {
+	return "Command Generation"
+}
+
+func (c *commandGenerationTool) Description() string {
+	// acknowledgement field is used to convince the LLM to return the JSON.
+	// Base on my testing LLM ignores the JSON when the schema has only one field.
+	// Adding additional "pseudo-fields" to the schema makes the LLM return the JSON.
+	return fmt.Sprintf(`Generate a Bash command.
+The input must be a JSON object with the following schema:
+%vjson
+{
+	"command": string, \\ The generated command
+	"acknowledgement": boolean \\ Set to true to ackowledge that you understand the formatting
+}
```

Comment:
I'm not too familiar with the structure of Assist but could this generate markdown instead? We could stream it to the UI quickly if that was the case, with the suggested command going in between three backticks

Reply:
OpenAI sends the markdown back most of the time.
https://github.com/gravitational/teleport/blob/88959155b46ce0bebb45da4525a8a58c768a8c28/lib/ai/model/prompt.go#L104